### ISSUE TITLE
Cursorless not aware of return types to a function

### ISSUE THREAD

**Issue by alexhwoods** [pokey/cursorless-vscode#442]
Having a scope that corresponds to the return type of a function would be useful

```typescript
function foo() {
  return "hello" // cursor at end of this line
}
```

“change return” (or something)

```typescript
function foo(): <cursor-here> {
  return "hello"
}
```

**Comment by pokey** [comment]
cc/ @AndreasArvidsson @soheeyang @sterlind

**Comment by AndreasArvidsson** [comment]
I'm unsure if I like the idea of clear actually adding text.

**Comment by pokey** [comment]
I have slightly mixed feelings as well. But the capability described here seems useful. Any ideas for alternative ways to express this command?

**Comment by AndreasArvidsson** [comment]
I agree that it's useful and I have had a need for something similar.

**Comment by soheeyang** [comment]
I'm sorry for the late catch-up! I also think that the command would be useful, but using `clear` for this function might not be an optimal solution.

**Comment by pokey** [comment]
So do we add a new "add" action then? "Set"?

**Comment by soheeyang** [comment]
Would there be more cases to _add_ something other than type? (Would the action be more generalized?)
Also, for the languages that do not have the type annotation, would the command be deactivated?

**Comment by AndreasArvidsson** [comment]
When the action is called `change` it makes perfect sense. As a `clear` command I'm a bit ambivalent. Also adding a new action for this very niche case is probably unnecessary.

**Comment by soheeyang** [comment]
My opinion is the same as Andreas's.

**Comment by pokey** [comment]
But the action is called "change" today

**Comment by AndreasArvidsson** [comment]
The default spoken form is. The action is called `clearAndSetSelection`

**Comment by pokey** [comment]
Yes maybe we were a bit too explicit there 😅

**Comment by AndreasArvidsson** [comment]
Yes if the action is going to be change on even add the action name is definitely incorrect
